test/parallel/test-http-pipeline-regr-2639.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+const http = require('http');
+
+var count = 0;
+const server = http.createServer(function(req, res) {
+  const id = count++;
+  setTimeout(function() {
+    res.end('hello');
+  }, (2 - id) * 10);
+
+  if (id === 1)
+    server.close();
+}).listen(common.PORT, function() {
```

Comment:
@Fishrock123 I think this is way too much :) If this is not going to work - we are pretty much in a bad situation. What about verifying the `count` instead?

Also, there is a possibly better fix for this: https://github.com/nodejs/node/pull/3068 .